integration/shared/isolated/marketplace_command_test.go

```diff
@@ -0,0 +1,336 @@
+								Eventually(session).Should(Say("FAILED"))
+								Eventually(session.Err).Should(Say("Service offering 'not-found-service' not found"))
+								Eventually(session).Should(Exit(1))
+							})
+						})
+
+						When("the specified service exists", func() {
+							var (
+								broker helpers.ServiceBroker
+								org    string
+								space  string
+							)
+
+							BeforeEach(func() {
+								org = helpers.NewOrgName()
+								space = helpers.NewSpaceName()
+								helpers.SetupCF(org, space)
+								helpers.TargetOrgAndSpace(org, space)
+
+								domain := helpers.DefaultSharedDomain()
+
+								broker = createBroker(domain, "SERVICE", "SERVICE-PLAN")
+								enableServiceAccess(broker)
+							})
+
+							AfterEach(func() {
+								broker.Destroy()
+								helpers.QuickDeleteOrg(org)
+							})
+
+							It("displays extended information about the service", func() {
+								description := "Shared fake Server, 5tb persistent disk, 40 max concurrent connections"
+								session := helpers.CF("marketplace", "-s", getServiceName(broker))
+								Eventually(session).Should(Say("Getting service plan information for service %s as %s\\.\\.\\.", getServiceName(broker), user))
+								Eventually(session).Should(Say("OK"))
+								Eventually(session).Should(Say("\n\n"))
+								Eventually(session).Should(Say("service plan\\s+description\\s+free or paid"))
+								Eventually(session).Should(Say("%s\\s+%s\\s+%s", getPlanName(broker), description, "free"))
+								Eventually(session).Should(Exit(0))
+							})
+						})
+
+						When("the specified service is accessible but not in the targeted space", func() {
+							var (
+								broker helpers.ServiceBroker
+								org    string
+								space  string
+							)
+
+							BeforeEach(func() {
+								org = helpers.NewOrgName()
+								space = helpers.NewSpaceName()
+								helpers.SetupCF(org, space)
+								helpers.TargetOrgAndSpace(org, space)
+
+								domain := helpers.DefaultSharedDomain()
+
+								broker = createBroker(domain, "SERVICE", "SERVICE-PLAN")
+								enableServiceAccessForOrg(broker, org)
+
+								helpers.TargetOrgAndSpace(ReadOnlyOrg, ReadOnlySpace)
+							})
+
+							AfterEach(func() {
+								helpers.TargetOrgAndSpace(org, space)
+								broker.Destroy()
+								helpers.QuickDeleteOrg(org)
+							})
+
+							It("displays an error that the service doesn't exist", func() {
+								session := helpers.CF("marketplace", "-s", getServiceName(broker))
+								Eventually(session).Should(Say("Getting service plan information for service %s as %s\\.\\.\\.", getServiceName(broker), user))
+								Eventually(session).Should(Say("FAILED"))
+								Eventually(session.Err).Should(Say("Service offering '%s' not found", getServiceName(broker)))
+								Eventually(session).Should(Exit(1))
+							})
+						})
+					})
+				})
+			})
+		})
+	})
+})
+
+func createBroker(domain, serviceName, planName string) helpers.ServiceBroker {
```

Comment:
Thanks @georgi-lozev. We noticed that this functionality is repeated in other parts of the codebase.
A few places that we saw are:
 * https://github.com/cloudfoundry/cli/blob/52a7bffaf9433a1e064def52984115ae0117971c/integration/shared/global/share_service_command_test.go#L129-L132 
* https://github.com/cloudfoundry/cli/blob/52a7bffaf9433a1e064def52984115ae0117971c/integration/shared/global/unshare_service_command_test.go#L126-L129

It will great if we could move this into a helper for code reusability and reduce duplication.  It's not required for this PR but it would be great in a future PR.

Reply:
Here is the [chore](https://www.pivotaltracker.com/story/show/161672768) in our backlog to follow-up on this, thanks again.